src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -117,13 +117,13 @@ trait TypeDiagnostics {
    */
   final def exampleTuplePattern(names: List[Name]): String = {
     val arity = names.length
-    val varPatterNames: Option[List[String]] = sequence(names map {
+    val varPatternNames: Option[List[String]] = sequence(names map {
```

Comment:
In context I think the `val` is about variables so it's okay to leave as-is,

```
    val varPatterNames: Option[List[String]] = sequence(names map {
      case name if nme.isVariableName(name) => Some(name.decode)
      case _                                => None
    })
```
WDYT?